File: Python/containsDuplicate.py
from typing import List

class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        return len(nums) != len(set(nums))
         

# Another Solution:
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        seen = set()  # using a set to store seen numbers
        for num in nums:
            if num in seen:
                return True
            seen.add(num)
        return False
         
        

File: JS/containsDuplicate.js
function containsDuplicate(nums) {
  let seen = new Set(); // Create a Set to store unique numbers

  for (let num of nums) {
    // Loop through each number in the array
    if (seen.has(num)) {
      // Check if the number is already in the set
      return true; // If yes, a duplicate exists
    }
    seen.add(num); // If no, add the number to the set
  }
  return false; // No duplicates found
}
